### PR TITLE
Add click(touch) handling feature

### PR DESCRIPTION
**Pull request by codelyne**
- add a new function handleClick to provide click(touch) handling

- now all the static values are stored in this.statics

- add a new function changeStates() as repeatedly used code

### CODE REVIEW

**Comment by codelyne** [comment]
I'm not sure about the name of "changeStates" function. It seems to me there is another better name based on its functionality.

**Comment by ubcent** [comment]
Hi, @codelyne your changes look great. Could you please add the "tap" event handling?

File: src/index.jsx [hunk 1]
```diff
@@ -78,24 +108,30 @@ class CircularColor extends PureComponent {
...
-        color: hsvToRgb(Math.abs(angle) * 180 / Math.PI)
-      });
+      this.changeStates(angle);
+    }
+  }
 
-      if(typeof onChange === 'function') {
-        onChange(hsvToRgb(Math.abs(angle) * 180 / Math.PI));
-      }
+  changeStates = (angle) => {
```
Comment (ubcent) [inline src/index.jsx]: As it seems to me, the better name of this method is something like "changeHandlerPosition"

Comment (ubcent) [reply src/index.jsx]: Or maybe "updateHandlerPosition"

File: src/index.jsx [hunk 2]
```diff
@@ -54,6 +61,29 @@ class CircularColor extends PureComponent {
...
+
+    const angle = -Math.atan2(y, x);
+
+    const outX = Math.abs(Math.floor((outerRadius) * Math.cos(angle)));
+    const outY = Math.abs(Math.floor((outerRadius) * Math.sin(angle)));
+
+    const inX = Math.abs(Math.floor((innerRadius) * Math.cos(angle)));
+    const inY = Math.abs(Math.floor((innerRadius) * Math.sin(angle)));
+
+    if (Math.abs(x) >= inX && Math.abs(x) <= outX
```
Comment (ubcent) [inline src/index.jsx]: I'm not sure about this. I'd recommend you should add something like "epsilon" for better "look and feel" on mobile devices

File: src/index.jsx (hunk 1 above)
Comment (codelyne) [reply src/index.jsx]: Just inside this function, we change the handler position as well as the color. Moreover, we call the callback from the parent component. Would it be better to divide this function into two different functions?

Comment (ubcent) [reply src/index.jsx]: Actually, I don't think so. In this case, it's overhead

File: src/index.jsx (hunk 2 above)
Comment (codelyne) [reply src/index.jsx]: Should this "epsilon" be inside the props of the component CircularColor? If yes, what the default value?

File: src/index.jsx [hunk 3]
```diff
@@ -63,7 +65,7 @@ class CircularColor extends PureComponent {
 
   handleClick = (event) => {
     const { size } = this.props;
-    const { innerRadius, outerRadius } = this.statics;
+    const { innerRadius, outerRadius, epsilon } = this.statics;
```
Comment (ubcent) [inline src/index.jsx]: I guess epsilon shouldn't be a class member. I'd recommend specifying it as a static property.

Comment (ubcent) [reply src/index.jsx]: Or it can be just a global variable

**Pull request merged**